public/javascripts/timeline.js

```diff
@@ -43,35 +50,115 @@ function update () {
+
+    } else if (slider_value <=3 && slider_value >=2) {
+
+             // Determine the pixel offset of the current-time-bar from the left
+            this.now_offset = .25 * hour_tick_list.offsetWidth + ((now - selectedDate) / MS_PER_HOUR ) * hour_width;
+            document.querySelector('#current-time-bar').style.left = this.now_offset + "px"
+
+            // Calculate the start point of the first hour
+            var hours_to_left = this.now_offset / this.hour_width;
+
+            var hour_start = Math.floor(now.getHours() - hours_to_left);
+            this.initial_offset = now_offset + ((hour2Date(hour_start) - now) / MS_PER_HOUR) * this.hour_width;
+            var hour_tick = createHourTick2(hour_start);
+     
+            hour_tick.style.left = initial_offset + "px";
+            hour_tick_list.appendChild(hour_tick);
+
+        for (var i = 1; i < num_hours + 1; i++) {
+            var hour_tick = createHourTick2(i + hour_start);
+            hour_tick.style.left = initial_offset + "px";
+            hour_tick_list.appendChild(hour_tick);
+        }
+
+        // console.log(hour_tick_list)
+
+        updateElemOffset('.item');
+        updateEventDuration();
+
+        // display minutes and hours
+
+    } else if (slider_value <=2 && slider_value >=1) {
+
+           slider_value = 1;
+
+        num_hours = Math.pow(slider_value, 2);
+        num_minutes = num_hours*60;
+        this.hour_width = hour_tick_list.offsetWidth / num_hours;
+        this.minute_width = hour_tick_list.offsetWidth / num_minutes;
+
+        this.now_offset = .25 * hour_tick_list.offsetWidth + ((now - selectedDate) / MS_PER_HOUR) * hour_width;
+        document.querySelector('#current-time-bar').style.left = this.now_offset + "px";
+
+        // Calculate the start point of the first hour
+        var hours_to_left = this.now_offset / this.hour_width;
+        // Calculate the start point of the first minute
+        var minutes_to_left = this.now_offset / this.minute_width;
+ 
+        var now = new Date();
+        hour_start = Math.floor(now.getHours() - hours_to_left);
+        minute_start = Math.floor(now.getMinutes() + 2 - minutes_to_left);
+        initial_offset = this.now_offset + ((minute2Date(minute_start) - now) / MS_PER_HOUR) * this.minute_width;
+        
+        // Create first minute tick and place it on timeline
+        var minute_tick = createMinuteTick(hour_start, minute_start);
+
+        minute_tick.style.left = initial_offset + "px";
+        hour_tick_list.appendChild(minute_tick); 
 
-    var hour_start = Math.floor(now.getHours() - hours_to_left);
-    this.initial_offset = now_offset + ((hour2Date(hour_start) - now) / MS_PER_HOUR) * this.hour_width;
+        for (var i = 1; i < num_minutes + 2; i++) {
+            var minute_tick = createMinuteTick(hour_start, i + minute_start);
+            minute_tick.style.left = initial_offset + "px";
+            hour_tick_list.appendChild(minute_tick);
 
```

Comment:
Looking through the above code, it seems like most of the code could be condensed into a single block with the if-statement simply reassigning a couple values as needed. To make the variable names line up, maybe the name hour-tick should simply be replaced with tick in most instances.

Doing so should reduce the duplication of the code dramatically and make it easy to pick out and modify the actual algorithms for determining when minute ticks appear and how they are spaced.